hyper/http11/connection.py

```diff
@@ -192,6 +204,23 @@ def request(self, method, url, body=None, headers=None):
 
         return
 
+    def _headers_to_http_header_map(self, headers):
+        headers = headers or {}
+        if not isinstance(headers, HTTPHeaderMap):
+            if isinstance(headers, Mapping):
+                headers = HTTPHeaderMap(headers.items())
+            elif isinstance(headers, Iterable):
+                headers = HTTPHeaderMap(headers)
+            else:
+                raise ValueError(
+                    'Header argument must be a dictionary or an iterable'
+                )
+        return headers
```

Comment:
For now, top of the file as a function. I'd consider a classmethod of `HTTPHeaderMap`, but let's save that for a follow-up PR rather than shoving it in this one, as both of us are already a bit nervous about the size of the diff. :wink:

Agreed that the two are inconsistent, but let's leave them inconsistent for now and fix them up later with a useful `HTTPHeaderMap` classmethod in a follow-up PR.